spec/services/gpo_confirmation_maker_spec.rb

```diff
@@ -1,68 +1,104 @@
+        attributes[key] = value
+      end
+      attributes
+    end
+    let(:profile) { create(:profile) }
 
-  subject { described_class.new(pii: pii, service_provider: service_provider, profile: profile) }
+    subject { described_class.new(pii: pii, service_provider: service_provider, profile: profile) }
 
-  describe '#perform' do
-    before do
-      allow(Base32::Crockford).to receive(:encode).and_return(otp)
-    end
+    describe '#perform' do
+      it 'should create a GpoConfirmation with the encrypted attributes' do
+        expect { subject.perform }.to change { GpoConfirmation.count }.from(0).to(1)
+
+        gpo_confirmation = GpoConfirmation.first
+        entry_hash = gpo_confirmation.entry
+        expect(entry_hash).to eq decrypted_attributes
+      end
 
-    it 'should create a GpoConfirmation with the encrypted attributes' do
-      expect { subject.perform }.to change { GpoConfirmation.count }.from(0).to(1)
+      it 'should create a GpoConfirmationCode with the profile and the encrypted OTP' do
+        expect { subject.perform }.to change { GpoConfirmationCode.count }.from(0).to(1)
 
-      gpo_confirmation = GpoConfirmation.first
-      entry_hash = gpo_confirmation.entry
-      expect(entry_hash).to eq decrypted_attributes
+        gpo_confirmation_code = GpoConfirmationCode.first
+
+        expect(gpo_confirmation_code.profile).to eq profile
+        expect(gpo_confirmation_code.otp_fingerprint).to eq Pii::Fingerprinter.fingerprint(otp)
+      end
     end
 
-    it 'should create a GpoConfirmationCode with the profile and the encrypted OTP' do
-      expect { subject.perform }.to change { GpoConfirmationCode.count }.from(0).to(1)
+    describe '#otp' do
+      it 'should return a normalized, 10 digit code' do
+        otp = subject.otp
+        expect(otp.length).to eq 10
+        expect(Base32::sCrockford.normalize(otp)).to eq otp
+      end
+
+      it 'filters out profane words' do
+        profane = Base32::Crockford.decode('FART')
+        not_profane = Base32::Crockford.decode('ABCD')
 
-      gpo_confirmation_code = GpoConfirmationCode.first
+        expect(SecureRandom).to receive(:random_number).
+          and_return(profane, not_profane)
 
-      expect(gpo_confirmation_code.profile).to eq profile
-      expect(gpo_confirmation_code.otp_fingerprint).to eq Pii::Fingerprinter.fingerprint(otp)
+        expect(subject.otp).to eq('000000ABCD')
+      end
     end
   end
 
-  describe '#otp' do
-    it 'should return a normalized, 10 digit code' do
-      otp = subject.otp
-
-      expect(otp.length).to eq 10
-      expect(Base32::Crockford.normalize(otp)).to eq otp
+  context "with a (bogus) zip+1" do
+    let(:otp) { '123ABC' }
+    let(:issuer) { 'this-is-an-issuer' }
+    let(:service_provider) { build(:service_provider, issuer: issuer) }
+    let(:decrypted_attributes) do
+      {
+        address1: '123 main st', address2: '',
+        city: 'Baton Rouge', state: 'Louisiana', zipcode: '12345+0',
```

Comment:
would it be possible to set up `let(:zipcode)` so we can override just the one value for this spec instead of having to set up all the context again?

Reply:
Done